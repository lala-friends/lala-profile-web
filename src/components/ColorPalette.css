.colorPalette {
    display: flex;
    flex-wrap: wrap;
    justify-content: space-around;
    margin-top: 0.5em;
}
.colorPalette .color {
    min-width: 2em;
    height: 2em;
    text-align: center;
}
