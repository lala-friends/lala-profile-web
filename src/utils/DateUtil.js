export const dateYYYYMMDD = (stringDate) => {
    const date = new Date(stringDate.toString());
    return date.getFullYear() + '-' + pad(date.getMonth()+1, 2) + '-' + pad(date.getDate(), 2);
};

export const dateYYYYMMDDHHMMSS = (stringDate) => {
    const date = new Date(stringDate.toString());
    return date.getFullYear() + '-' + pad(date.getMonth()+1, 2) + '-' + pad(date.getDate(), 2) + ' ' +
        pad(date.getHours(), 2) + ':' + pad(date.getMinutes(), 2) + ':' + pad(date.getSeconds(), 2);
}

const pad = (number, width) => {
    let numberStr = number.toString();
    if (numberStr.length >= width) {
        return numberStr;
    } else {
        for(let i=width-numberStr.length; i>0; i--)
            numberStr = '0' + numberStr;
        return numberStr;
    }
};