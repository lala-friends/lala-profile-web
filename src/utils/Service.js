import axios from 'axios';
import {baseURL} from '../config';

class Service {
    constructor() {
        const service = axios.create({
            baseURL: baseURL
        });
        this.service = service;
    }

    get(path, callback) {
        return this.service.get(path).then(
            (response) => callback(response.status, response.data)
        );
    }

    post(path, payload, callback) {
        return this.service.post(path, payload).then(
            (response) => callback(response.status, response.data)
        );
    }

    put(path, payload, callback) {
        return this.service.put(path, payload).then(
            (response) => callback(response.status, response.data)
        );
    }

    delete(path, callback) {
        return this.service.delete(path).then(
            (response) => callback(response.status, response.data)
        );
    }
}

export default Service;