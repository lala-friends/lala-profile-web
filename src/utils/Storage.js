import firebase from 'firebase';
import {firebaseConfig} from '../config';
firebase.initializeApp(firebaseConfig);

export const uploadImage = (file) => {
    const task = firebase.storage().ref().child(file.name).put(file, { contentType: file.type });
    return task;
}