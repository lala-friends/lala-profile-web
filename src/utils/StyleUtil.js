export const rgbToHexa = (rgb) => {
    return getHexa(...rgb.substring(4, rgb.length-1).split(', '));
}
const getHexa = (red, green, blue) => {
    const rgb = blue | (green << 8) | (red << 16);
    return '#' + (0x1000000 + rgb).toString(16).slice(1)
};

export const colorPalette = ['#ffa8a8', '#faa2c1', '#eebefa', '#d0bfff', '#bac8ff', '#a5d8ff', '#c5f6fa', '#99e9f2', '#96f2d7', '#b2f2bb', '#ffe066', '#ff922b'];