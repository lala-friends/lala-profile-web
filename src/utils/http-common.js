import axios from 'axios';
import https from 'https';
import {baseURL} from "../config";

export default axios.create({
    baseURL: baseURL,
    httpsAgent: new https.Agent({
        rejectUnauthorized: false
    }),
});
