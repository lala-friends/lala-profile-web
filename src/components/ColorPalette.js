import React from 'react';
import PropTypes from 'prop-types';
import {colorPalette} from '../utils/StyleUtil';

const ColorPalette = (props) => {
    return (
        <div>
            {colorPalette.map((color, key) => {
                return (
                    <div key={key} className='color' style={{backgroundColor: `${color}`}} onClick={props.changeColor}>
                        {
                            (props.color===color) && (<i className='mdi mdi-check' />)
                        }
                    </div>)
            })}
        </div>
    )
}

ColorPalette.propTypes = {
    color: PropTypes.string.isRequired,
    changeColor: PropTypes.func.isRequired
}

export default ColorPalette;