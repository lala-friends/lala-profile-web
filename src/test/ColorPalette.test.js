import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
// React 16 Enzyme adapter
Enzyme.configure({ adapter: new Adapter() });

import ColorPalette from '../components/ColorPalette';
import {colorPalette} from '../utils/StyleUtil';

describe('ColorPalette', () => {
    test('ColorPalette이 렌더링 된다.', () => {
        const changeColor = jest.fn();
        const colorIndex = 0;
        const selectedColor = colorPalette[colorIndex];

        const wrapper = shallow(<ColorPalette color={selectedColor} changeColor={changeColor}/>);
         
        expect(wrapper).toMatchSnapshot();
        // 선택된 color에 check표시 되어 렌더링된다.
        expect(wrapper.find('.color').get(colorIndex).props.children.props.className).toContain('mdi-check');
    });

    test('다른 color를 click하면, props.changeColor가 호출된다.', () => {
        const changeColor = jest.fn();
        const colorIndex = 0;
        const selectedColor = colorPalette[colorIndex];

        const wrapper = shallow(<ColorPalette color={selectedColor} changeColor={changeColor}/>);

        const selectedIndex = 1;
        wrapper.find('.color').at(selectedIndex).simulate('click');
        expect(changeColor).toHaveBeenCalled();
    });
});