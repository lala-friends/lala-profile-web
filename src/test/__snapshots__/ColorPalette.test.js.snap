// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPalette ColorPalette이 렌더링 된다. 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ColorPalette
    changeColor={[Function]}
    color="#ffa8a8"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="color"
          onClick={[Function]}
          style={
                    Object {
                              "backgroundColor": "#ffa8a8",
                            }
          }
>
          <i
                    className="mdi mdi-check"
          />
</div>,
        <div
          className="color"
          onClick={[Function]}
          style={
                    Object {
                              "backgroundColor": "#faa2c1",
                            }
          }
/>,
        <div
          className="color"
          onClick={[Function]}
          style={
                    Object {
                              "backgroundColor": "#eebefa",
                            }
          }
/>,
        <div
          className="color"
          onClick={[Function]}
          style={
                    Object {
                              "backgroundColor": "#d0bfff",
                            }
          }
/>,
        <div
          className="color"
          onClick={[Function]}
          style={
                    Object {
                              "backgroundColor": "#bac8ff",
                            }
          }
/>,
        <div
          className="color"
          onClick={[Function]}
          style={
                    Object {
                              "backgroundColor": "#a5d8ff",
                            }
          }
/>,
        <div
          className="color"
          onClick={[Function]}
          style={
                    Object {
                              "backgroundColor": "#c5f6fa",
                            }
          }
/>,
        <div
          className="color"
          onClick={[Function]}
          style={
                    Object {
                              "backgroundColor": "#99e9f2",
                            }
          }
/>,
        <div
          className="color"
          onClick={[Function]}
          style={
                    Object {
                              "backgroundColor": "#96f2d7",
                            }
          }
/>,
        <div
          className="color"
          onClick={[Function]}
          style={
                    Object {
                              "backgroundColor": "#b2f2bb",
                            }
          }
/>,
        <div
          className="color"
          onClick={[Function]}
          style={
                    Object {
                              "backgroundColor": "#ffe066",
                            }
          }
/>,
        <div
          className="color"
          onClick={[Function]}
          style={
                    Object {
                              "backgroundColor": "#ff922b",
                            }
          }
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": <i
            className="mdi mdi-check"
/>,
          "className": "color",
          "onClick": [Function],
          "style": Object {
            "backgroundColor": "#ffa8a8",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "mdi mdi-check",
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": false,
          "className": "color",
          "onClick": [Function],
          "style": Object {
            "backgroundColor": "#faa2c1",
          },
        },
        "ref": null,
        "rendered": false,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": false,
          "className": "color",
          "onClick": [Function],
          "style": Object {
            "backgroundColor": "#eebefa",
          },
        },
        "ref": null,
        "rendered": false,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "host",
        "props": Object {
          "children": false,
          "className": "color",
          "onClick": [Function],
          "style": Object {
            "backgroundColor": "#d0bfff",
          },
        },
        "ref": null,
        "rendered": false,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "4",
        "nodeType": "host",
        "props": Object {
          "children": false,
          "className": "color",
          "onClick": [Function],
          "style": Object {
            "backgroundColor": "#bac8ff",
          },
        },
        "ref": null,
        "rendered": false,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "5",
        "nodeType": "host",
        "props": Object {
          "children": false,
          "className": "color",
          "onClick": [Function],
          "style": Object {
            "backgroundColor": "#a5d8ff",
          },
        },
        "ref": null,
        "rendered": false,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "6",
        "nodeType": "host",
        "props": Object {
          "children": false,
          "className": "color",
          "onClick": [Function],
          "style": Object {
            "backgroundColor": "#c5f6fa",
          },
        },
        "ref": null,
        "rendered": false,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "7",
        "nodeType": "host",
        "props": Object {
          "children": false,
          "className": "color",
          "onClick": [Function],
          "style": Object {
            "backgroundColor": "#99e9f2",
          },
        },
        "ref": null,
        "rendered": false,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "8",
        "nodeType": "host",
        "props": Object {
          "children": false,
          "className": "color",
          "onClick": [Function],
          "style": Object {
            "backgroundColor": "#96f2d7",
          },
        },
        "ref": null,
        "rendered": false,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "9",
        "nodeType": "host",
        "props": Object {
          "children": false,
          "className": "color",
          "onClick": [Function],
          "style": Object {
            "backgroundColor": "#b2f2bb",
          },
        },
        "ref": null,
        "rendered": false,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "10",
        "nodeType": "host",
        "props": Object {
          "children": false,
          "className": "color",
          "onClick": [Function],
          "style": Object {
            "backgroundColor": "#ffe066",
          },
        },
        "ref": null,
        "rendered": false,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "11",
        "nodeType": "host",
        "props": Object {
          "children": false,
          "className": "color",
          "onClick": [Function],
          "style": Object {
            "backgroundColor": "#ff922b",
          },
        },
        "ref": null,
        "rendered": false,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="color"
            onClick={[Function]}
            style={
                        Object {
                                    "backgroundColor": "#ffa8a8",
                                  }
            }
>
            <i
                        className="mdi mdi-check"
            />
</div>,
          <div
            className="color"
            onClick={[Function]}
            style={
                        Object {
                                    "backgroundColor": "#faa2c1",
                                  }
            }
/>,
          <div
            className="color"
            onClick={[Function]}
            style={
                        Object {
                                    "backgroundColor": "#eebefa",
                                  }
            }
/>,
          <div
            className="color"
            onClick={[Function]}
            style={
                        Object {
                                    "backgroundColor": "#d0bfff",
                                  }
            }
/>,
          <div
            className="color"
            onClick={[Function]}
            style={
                        Object {
                                    "backgroundColor": "#bac8ff",
                                  }
            }
/>,
          <div
            className="color"
            onClick={[Function]}
            style={
                        Object {
                                    "backgroundColor": "#a5d8ff",
                                  }
            }
/>,
          <div
            className="color"
            onClick={[Function]}
            style={
                        Object {
                                    "backgroundColor": "#c5f6fa",
                                  }
            }
/>,
          <div
            className="color"
            onClick={[Function]}
            style={
                        Object {
                                    "backgroundColor": "#99e9f2",
                                  }
            }
/>,
          <div
            className="color"
            onClick={[Function]}
            style={
                        Object {
                                    "backgroundColor": "#96f2d7",
                                  }
            }
/>,
          <div
            className="color"
            onClick={[Function]}
            style={
                        Object {
                                    "backgroundColor": "#b2f2bb",
                                  }
            }
/>,
          <div
            className="color"
            onClick={[Function]}
            style={
                        Object {
                                    "backgroundColor": "#ffe066",
                                  }
            }
/>,
          <div
            className="color"
            onClick={[Function]}
            style={
                        Object {
                                    "backgroundColor": "#ff922b",
                                  }
            }
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": <i
              className="mdi mdi-check"
/>,
            "className": "color",
            "onClick": [Function],
            "style": Object {
              "backgroundColor": "#ffa8a8",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "mdi mdi-check",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": false,
            "className": "color",
            "onClick": [Function],
            "style": Object {
              "backgroundColor": "#faa2c1",
            },
          },
          "ref": null,
          "rendered": false,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": false,
            "className": "color",
            "onClick": [Function],
            "style": Object {
              "backgroundColor": "#eebefa",
            },
          },
          "ref": null,
          "rendered": false,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "host",
          "props": Object {
            "children": false,
            "className": "color",
            "onClick": [Function],
            "style": Object {
              "backgroundColor": "#d0bfff",
            },
          },
          "ref": null,
          "rendered": false,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "host",
          "props": Object {
            "children": false,
            "className": "color",
            "onClick": [Function],
            "style": Object {
              "backgroundColor": "#bac8ff",
            },
          },
          "ref": null,
          "rendered": false,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "5",
          "nodeType": "host",
          "props": Object {
            "children": false,
            "className": "color",
            "onClick": [Function],
            "style": Object {
              "backgroundColor": "#a5d8ff",
            },
          },
          "ref": null,
          "rendered": false,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "6",
          "nodeType": "host",
          "props": Object {
            "children": false,
            "className": "color",
            "onClick": [Function],
            "style": Object {
              "backgroundColor": "#c5f6fa",
            },
          },
          "ref": null,
          "rendered": false,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "7",
          "nodeType": "host",
          "props": Object {
            "children": false,
            "className": "color",
            "onClick": [Function],
            "style": Object {
              "backgroundColor": "#99e9f2",
            },
          },
          "ref": null,
          "rendered": false,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "8",
          "nodeType": "host",
          "props": Object {
            "children": false,
            "className": "color",
            "onClick": [Function],
            "style": Object {
              "backgroundColor": "#96f2d7",
            },
          },
          "ref": null,
          "rendered": false,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "9",
          "nodeType": "host",
          "props": Object {
            "children": false,
            "className": "color",
            "onClick": [Function],
            "style": Object {
              "backgroundColor": "#b2f2bb",
            },
          },
          "ref": null,
          "rendered": false,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "10",
          "nodeType": "host",
          "props": Object {
            "children": false,
            "className": "color",
            "onClick": [Function],
            "style": Object {
              "backgroundColor": "#ffe066",
            },
          },
          "ref": null,
          "rendered": false,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "11",
          "nodeType": "host",
          "props": Object {
            "children": false,
            "className": "color",
            "onClick": [Function],
            "style": Object {
              "backgroundColor": "#ff922b",
            },
          },
          "ref": null,
          "rendered": false,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
